In [2]:
from llama_parse import LlamaParse
from dotenv import load_dotenv
import os

load_dotenv()

LLAMA_PARSE_API = os.getenv("LLAMA_PARSE_API")


parser = LlamaParse(
    api_key=LLAMA_PARSE_API,
    parse_mode="parse_page_with_llm",
    result_type="markdown",
    high_res_ocr=True,
    adaptive_long_table=True,  # Adaptive long table. LlamaParse will try to detect long table and adapt the output
    outlined_table_extraction=True, 
)


pages = parser.load_data("Uni-Verse-Files/syllabi/SyllabusDiscrete structures2026.docx")

Started parsing the file under job_id 60ac3867-df37-4908-92d8-b4d8ad91ba17


In [3]:
pages

[Document(id_='9459431e-f9c7-4536-b6a5-a3931ce94f34', embedding=None, metadata={}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={}, metadata_template='{key}: {value}', metadata_separator='\n', text_resource=MediaResource(embeddings=None, data=None, text='\nKAZAKH-BRITISH TECHNICAL UNIVERSITY\nSCHOOL OF APPLIED MATHEMATICS\nApproved by\nDean of School of\nApplied Mathematics\n___________ Kenzhebayev T.S.\n«13 » January 2026\n\n# Syllabus\n\n# Discrete Structures CSCI 1102\n\n# Semester: Spring 2026\n\n# 2025/2026 Academic Year\n\n# 3 credits (2/0/1)\n\n# Instructor: Mustafayeva Ardak Bolatkyzy, senior lecturer\n\n| Personal Information about the Instructor | Time and place of classes | Contact information                     |\n| ----------------------------------------- | ------------------------- | --------------------------------------- |\n| MSc, senior lecturer                      | According to the schedule | Tel.: TBA e-mail: a.mustafaeva\\@kbtu.k

In [4]:
for page in pages:
    print(page.text)


KAZAKH-BRITISH TECHNICAL UNIVERSITY
SCHOOL OF APPLIED MATHEMATICS
Approved by
Dean of School of
Applied Mathematics
___________ Kenzhebayev T.S.
«13 » January 2026

# Syllabus

# Discrete Structures CSCI 1102

# Semester: Spring 2026

# 2025/2026 Academic Year

# 3 credits (2/0/1)

# Instructor: Mustafayeva Ardak Bolatkyzy, senior lecturer

| Personal Information about the Instructor | Time and place of classes | Contact information                     |
| ----------------------------------------- | ------------------------- | --------------------------------------- |
| MSc, senior lecturer                      | According to the schedule | Tel.: TBA e-mail: a.mustafaeva\@kbtu.kz |

# COURSE DURATION:

3 credits, 15 weeks, 45 class hours.

# COURSE PRE-REQUISITES:

Good knowledge of school mathematics is enough.

# COURSE DESCRIPTION

# Course objectives

Discrete Structures is a transfer-oriented course designed to meet the requirements of Computer and Information Science degree prog

KAZAKH-BRITISH TECHNICAL UNIVERSITY
SCHOOL OF APPLIED MATHEMATICS
Approved by
Dean of School of
Applied Mathematics
___________ Kenzhebayev T.S.
«13 » January 2026

# Syllabus

# Discrete Structures CSCI 1102

# Semester: Spring 2026

# 2025/2026 Academic Year

# 3 credits (2/0/1)

# Instructor: Mustafayeva Ardak Bolatkyzy, senior lecturer

| Personal Information about the Instructor | Time and place of classes | Contact information                     |
| ----------------------------------------- | ------------------------- | --------------------------------------- |
| MSc, senior lecturer                      | According to the schedule | Tel.: TBA e-mail: a.mustafaeva\@kbtu.kz |

# COURSE DURATION:

3 credits, 15 weeks, 45 class hours.

# COURSE PRE-REQUISITES:

Good knowledge of school mathematics is enough.

# COURSE DESCRIPTION

# Course objectives

Discrete Structures is a transfer-oriented course designed to meet the requirements of Computer and Information Science degree programs. Furthermore, this course is designed to meet the following program objectives. Upon successful completion of this course, students will be able to:

- Demonstrate critical thinking, analytical reasoning, and problem-solving skills
- Apply appropriate mathematical concepts and operations to interpret data and to solve problems
- Identify a problem and analyze it in terms of its significant parts and the information needed to solve it.
- Formulate and evaluate possible solutions to problems, and select and defend the chosen solutions.

# Course outcomes

At the end of the course, students are expected to:

- create compound statements, expressed in mathematical symbols or in English, to determine the truth or falseness of compound statements and to use the rules of inference to prove a conclusion statement from hypothesis statements by applying the rules of propositional and predicate calculus logic;
- prove mathematical statements involving numbers by applying various proof methods, which are based on the rules of inference from logic;
- prove the validity of sequences and series and the correctness of repeated processes by applying mathematical induction;
- define and identify the terms, rules, and properties of set theory and use these as tools to support problem solving and reasoning in applications of logic, functions, number theory, sequences, counting, trees and graphs, and finite state machines;
- solve recursive problems by applying knowledge of recursive sequences;
- create graphs and trees to represent and help prove or disprove statements, to make decisions or select from alternative choices, to calculate probabilities, to document derivation steps, or to solve problems;





# Course Overview

Construct and analyze finite state automata (another name for machines), formal languages, and regular expressions.

# Knowledge

During the study of this course, students must obtain knowledge about how to:

- Explain with examples the basic terminology of functions, relations, and sets;
- Perform the operations associated with sets, functions, and relations;
- Convert logical statements from informal language to propositional and predicate logic expressions;
- Apply formal methods of symbolic propositional and predicate logic, such as calculating validity of formulae and computing normal forms;
- Identify the proof technique used in a given proof;
- Determine which type of proof is best for a given problem;
- Explain the parallels between ideas of mathematical and/or structural induction to recursion and recursively defined structures;
- Explain the relationship between weak and strong induction and give examples of the appropriate use of each;
- State the well-ordering principle and its relationship to mathematical induction;
- Apply counting arguments, including sum and product rules, inclusion-exclusion principle and arithmetic/geometric progressions;
- Apply the pigeonhole principle in the context of a formal proof;
- Be familiar with elementary concepts of Languages and Automata theory and understand their role in Computer Science.

# Skills

As a result of studying this course, students must be able to:

- Apply formal logic proofs and/or informal, but rigorous, logical reasoning to real problems, such as predicting the behavior of software or solving problems such as puzzles;
- Demonstrate comprehension of discrete structures and their relevance within the context of computer science, in the areas of data structures and algorithms, in particular;
- Apply discrete structures into other computing problems such as formal specification, verification, databases, artificial intelligence, and cryptography;
- Demonstrate mathematical skills, analytical and critical thinking abilities;
- Communicate clearly and effectively using the technical language of the field correctly.

# Literature

# Required

1. Kenneth Rosen: “Discrete Mathematics and Its Applications”, 7th edition, 2012, McGraw-Hill Press, 2009.
2. Seymour Lipschutz, Marc Lipson: Discrete Mathematics. (Shaum’s Outline), 4th edition, 2009, McGraw-Hill.

# Supplementary

1. Discrete Mathematics, 2nd Edition by Biggs, Oxford University Press, 2013, Paperback, 9780198713692.

# Course Calendar

| Week | Topic                                                                | Class work            | SIW (student’s individual work)                                                                                               |
| ---- | -------------------------------------------------------------------- | --------------------- | ----------------------------------------------------------------------------------------------------------------------------- |
| 1    | Sets, Relations, and Functions.                                      | 2 Lectures, 1 Seminar | Recommended homeworks at the end of each part of the lectures (Deadline – Week 4, demonstrate your knowledge during the quiz) |
| 2    | Sets, Relations, and Functions (continuation). Proofs and Induction. | 2 Lectures, 1 Seminar | Recommended homeworks at the end of each part of the lectures (Deadline – Week 4, demonstrate your knowledge during the quiz) |
| 3    | Introduction to Number Theory.                                       | 2 Lectures, 1 Seminar | Recommended homeworks at the end of each part of the lectures (Deadline – Week 4, demonstrate your knowledge during the quiz) |






# 1. Introduction to Number Theory (continuation)

Recommended homeworks at the end of each part of the lectures (Deadline – Week 7, demonstrate your knowledge during the quiz)

# 2. Counting. Elements of Combinatorics

Recommended homeworks at the end of each part of the lectures (Deadline – Week 7, demonstrate your knowledge during the quiz)

# 3. Logic and Propositional Calculus

Recommended homeworks at the end of each part of the lectures (Deadline – Week 7, demonstrate your knowledge during the quiz)

# 4. Logic and Propositional Calculus (continuation)

Recommended homeworks at the end of each part of the lectures (Deadline – Week 11, demonstrate your knowledge during the quiz)

# 5. Boolean Algebra

Recommended homeworks at the end of each part of the lectures (Deadline – Week 11, demonstrate your knowledge during the quiz)

# 6. Boolean Algebra (continuation)

Recommended homeworks at the end of each part of the lectures (Deadline – Week 11, demonstrate your knowledge during the quiz)

# 7. Advanced Counting Techniques

Recommended homeworks at the end of each part of the lectures (Deadline – Week 11, demonstrate your knowledge during the quiz)

# 8. Elements of Graph Theory

Recommended homeworks at the end of each part of the lectures (Deadline – Week 14, demonstrate your knowledge during the quiz)

# 9. Elements of Graph Theory (continuation)

Recommended homeworks at the end of each part of the lectures (Deadline – Week 14, demonstrate your knowledge during the quiz)

# 10. Binary Trees

Recommended homeworks at the end of each part of the lectures (Deadline – Week 14, demonstrate your knowledge during the quiz)

# 11. Languages, Automata, Grammars

Recommended homeworks at the end of each part of the lectures (Deadline – Week 14, demonstrate your knowledge during the quiz)

# 12. Languages, Automata, Grammars (continuation)

# COURSE ASSESSMENT PARAMETERS

| Attendance and activity on lectures | 0%  |
| ----------------------------------- | --- |
| HomeWorks and SIW                   | 10% |
| Control works and midterm           | 40% |
| Seminar lessons                     | 10% |





# Final exam

40%

Total 100%

| No    | Assessment criteria                 | Weeks | 1 | 2  | 3  | 4  | 5 | 6  | 7 | 8 | 9 | 10 | 11   | 12 | 13 | 14    | 15 | Total |
| ----- | ----------------------------------- | ----- | - | -- | -- | -- | - | -- | - | - | - | -- | ---- | -- | -- | ----- | -- | ----- |
| 1.    | Attendance and activity on lectures | 1     | 1 | 1  | 1  | 1  | 1 | 1  | 1 | 1 | 1 | 1  | 1    | 1  | 1  | 15    |    |       |
| 2.    | Homeworks and SIW                   | 1     | 1 | 1  |    | 1  | 1 | 1  | 1 | 1 |   |    | 10   |    |    |       |    |       |
| 3.    | Control works and midterm           |       |   | 10 | 10 | 10 |   | 10 |   |   |   |    | 40.0 |    |    |       |    |       |
| 4.    | Activity on seminar lessons         |       |   |    |    |    |   |    |   |   |   |    |      |    |    |       |    |       |
| 6.    | Final examination                   |       |   |    |    |    |   |    |   |   |   |    |      |    | 40 | 40.0  |    |       |
| Total |                                     |       |   |    |    |    |   |    |   |   |   |    |      | 60 | 40 | 100.0 |    |       |

Lectures are conducted in the form of explaining the theory given in the course that is why students supplied with handouts uploaded into the intranet. Activity and attendance on lessons is mandatory. Mandatory requirement is preparation for each lesson.

# Grading policy:

Intermediate attestations (on 8th and 15th week) join topics of all lectures, laboratories, homework, quiz and materials for reading discussed to the time of attestation. Maximum number of points within attendance, activity, homework, quiz and laboratories for each attestation is 30 points.

Final exam joins and generalizes all course materials, is conducted in the complex form with questions and problems. Final exam duration is 120 min. Maximum number of points is 40. At the end of the semester you receive overall total grade (summarized index of your work during semester) according to conventional KBTU grade scale.

# ACADEMIC POLICY

Students are required:

- to be respectful to the teacher and other students;
- to switch off mobile phones during classes;
- DO NOT cheat. Plagiarized papers shall be graded with zero points!
- to come to classes prepared and actively participate in classroom work; to meet the deadlines;
- to enter the room before the teacher starts the lesson;
- to attend all classes. No make-up tests or quiz are allowed unless there is a valid reason for missing it;
- to follow KBTU academic policy regarding W, AW, I, F grades.
- When students are absent for 20% of the lessons or more (without Spravka), then their grade is F.
- When students have a score of 29 or less for attestation 1 added to attestation 2, then their grade is F.
- When students have a score of 19 or less (less than 50%) for their final exam, then their grade is F.
- When students do not come for their final exam, then their grade is F.

Students are encouraged to:

- consult the teacher on any issues related to the course;
- make up within a week’s time for the works undone for a valid reason without any grade deductions;

Senior lecturer of the School of Applied Mathematics Mustafayeva Ardak

Approved by protocol № 5 of the 13th of January, 2026

In [1]:
from parser_docx import DocxParser

parser = DocxParser("Uni-Verse-Files/syllabi/SyllabusDiscrete structures2026.docx")

result = parser.parse()

result

ParsedDoc(paragraphs=[Para(runs=[Run(text='           ', bold=True, italic=False, underline=False, font_name=None, font_size=11.0, color=None), Run(text='KAZAKH-BRITISH TECHNICAL UNIVERSITY', bold=True, italic=False, underline=False, font_name=None, font_size=11.0, color=None)], style='Normal', alignment='CENTER (1)', level=None), Para(runs=[Run(text='SCHOOL OF APPLIED MATHEMATICS', bold=True, italic=False, underline=False, font_name=None, font_size=11.0, color=None)], style='Normal', alignment='CENTER (1)', level=None), Para(runs=[], style='Normal', alignment='CENTER (1)', level=None), Para(runs=[Run(text='Approved by', bold=True, italic=False, underline=False, font_name=None, font_size=None, color=None)], style='Default', alignment=None, level=None), Para(runs=[Run(text='D', bold=True, italic=False, underline=False, font_name=None, font_size=None, color=None), Run(text='ean of School of ', bold=True, italic=False, underline=False, font_name=None, font_size=None, color=None)], style='

[HumanMessage(content=[{'type': 'text', 'text': 'Ниже — предложения по улучшению и аккуратно отредактированная версия текста (можете вставить её вместо исходной). Я ссылался на элементы макета и скриншоты по номерам страниц файла.\n\nКлючевые рекомендации (общие)\n- Единый стиль:\n  - Используйте «Шаг 1/2/3…» вместо «Первый/Второй…».\n  - Приведите к единому регистру заголовки: «Шаги», «Дополнительно», «Контакты».\n  - Последовательно применяйте кавычки «…» и длинное тире —.\n  - Пишите «учебный год 2025/2026, весенний семестр» (а не «год 2025–2026, семестр Весенний»).\n  - «Google Таблицы» вместо «Google Таблица».\n  - Уберите разговорные слова и местоимение «я» в гайде. Замените на нейтральный тон: «рекомендуется», «можно использовать».\n- Точность терминов:\n  - Поясните 1/60: «записано/максимум мест» (а не просто «количество зарегистрированных студентов»).\n  - Слово «вакансии» может быть непонятно. Лучше «свободные слоты/потоки (без закреплённого преподавателя)».\n  - Вместо «216а